lib/Conversion/TorchToLinalg/DataMovement.cpp

```diff
@@ -171,439 +172,348 @@ class ConvertAtenFlattenUsingIntsOp
       return failure();
-    Location loc = op.getLoc();
-    Value input = adaptor.getSelf();
-    auto inputType = input.getType().cast<RankedTensorType>();
-    SmallVector<Value> inputSize = getTensorSizes(rewriter, loc, input);
-    int64_t inputRank = inputType.getRank();
-    const TypeConverter *typeConverter = getTypeConverter();
-    auto resultType =
-        typeConverter->convertType(op.getType()).cast<RankedTensorType>();
-    int64_t resultRank = resultType.getRank();
-    if (resultRank == 0)
-      return rewriter.notifyMatchFailure(op,
-                                         "result shape of rank 0 is invalid");
+    Location loc = viewOp.getLoc();
 
-    // TODO: add support for case inputRank 0 expanded to size 1
-    if (inputRank == 0)
-      return rewriter.notifyMatchFailure(
-          op, "unimplemented: input rank 0 is not supported");
+    const TypeConverter *typeConverter = getTypeConverter();
 
-    // Extract the desired output size as a list of integers. This list should
-    // have been created using the operation `torch.prim.ListConstruct`.
-    SmallVector<Value> outputSizeTorchInt;
-    if (!getListConstructElements(op.getSize(), outputSizeTorchInt)) {
-      return rewriter.notifyMatchFailure(op,
+    Value input = adaptor.getSelf();
+    auto inType = input.getType().cast<RankedTensorType>();
+    SmallVector<Value> inSize = getTensorSizes(rewriter, loc, input);
+    int64_t inRank = inType.getRank();
+
+    auto outType =
+        typeConverter->convertType(viewOp.getType()).cast<RankedTensorType>();
+    int64_t outRank = outType.getRank();
+    SmallVector<Value> outSizeTorchInt;
+    if (!getListConstructElements(viewOp.getSize(), outSizeTorchInt)) {
+      return rewriter.notifyMatchFailure(viewOp,
                                          "unimplemented: the target size is "
                                          "not constructed from ListConstruct");
     }
-    if (llvm::count_if(outputSizeTorchInt, [](Value size) -> bool {
-          int64_t sizeInt;
-          if (matchPattern(size, m_TorchConstantInt(&sizeInt)))
-            return sizeInt == -1;
-          return false;
-        }) > 1) {
+    SmallVector<Value> outSizeInt =
+        getTypeConvertedValues(rewriter, loc, typeConverter, outSizeTorchInt);
+    if (outRank != static_cast<int64_t>(outSizeInt.size())) {
       return rewriter.notifyMatchFailure(
-          op, "at most one element in size list is allowed to be -1");
+          viewOp, "size list length mismatches with the result type rank");
     }
-    SmallVector<Value> outputSizeInt = getTypeConvertedValues(
-        rewriter, loc, typeConverter, outputSizeTorchInt);
-    if (resultRank != (int64_t)outputSizeInt.size()) {
+
+    if (inRank == 0 || outRank == 0) {
       return rewriter.notifyMatchFailure(
-          op, "desired size list length mismatches with the result type rank");
+          viewOp, "It is impossible to expand from or collapse to a scalar");
```

Comment:
This is not true. It is valid to have a rank-zero inputs and outputs (they are just not supported in this pattern yet):

```python
>>> import torch
>>> torch.ops.aten.view(torch.tensor(1), [1,1,1])
tensor([[[1]]])
>>> torch.ops.aten.view(torch.tensor(1), [])
tensor(1)
```

Reply:
For these cases, the error should be prefixed with `unimplemented: ...`. This means, the behavior is valid in PyTorch, but in torch-mlir we don't yet know how to handle it